lola_adeyemi_final_exam.py

```diff
@@ -0,0 +1,198 @@
+
+SEE NOTES BELOW
+
+"""
+
+from random import randint
+
+
+class CFGStudent:
+
+    def __init__(self, name, surname, age, email, student_id=None):
+        self.name = name
+        self.surname = surname
+        self.age = age
+        self.email = email
+        self.student_id = self.get_id()
+
+    @staticmethod
+    def generate_id():
+        random_id = randint(1000, 10000) #create new student id using random
+        return random_id
+
+    def get_id(self):
+        return self.generate_id()
+
+    def get_fullname(self):
+        create_fullname = (self.name.title()+" "+ self.surname.title())
+        return print(create_fullname)
+
+
+class NanoStudent(CFGStudent):
+
+    def __init__(self, name, surname, email, age, specialization, course_grades=None):
+        super().__init__(name, surname, email, age,)
+        self.specialization = specialization
+        self.course_grades = {course_grades}
+
+    @staticmethod
+    def generate_id():
+        random_id = randint(1000, 10000) #create new student id using random
+        id = print("NANO{}".format(random_id))
+        return id
+        #"Example 'NANO1234' "
+
+    def add_new_grade(self, subject, grade):
+        self.course_grades.add(subject=grade)
+
+    def get_course_grades(self):
+        return self.course_grades
+
+
+
+
+############################################
+#
+# # # Example run
+#
+s = CFGStudent('Anna', 'Smith', 18, 'anna@mail.com')  # do not pass ID, it should be generated automatically
+print(s.get_fullname())
+# returns 'Anna Smith'
+print(s.student_id)
+# returns '3868' or some other random number
+
+cfg_s = NanoStudent('Software', 'Mia', 'Papandopulu', 20, 'mia@mail.com')
+print(cfg_s.get_fullname())
+# returns 'Mia Papandopulu'
+print(cfg_s.get_id())
+# returns 'NANO6180' or some other random number
+
+cfg_s.add_new_grade('theory', 95)
+cfg_s.add_new_grade('project', 78)
+print(cfg_s.get_course_grades())
+#returns {'theory': 95, 'project': 78}
+
+#
+#
+# """
+# TASK 2
+#
+# Given an index limit, find all corresponding Fibonacci values up to that limit in a sequence
+# and return the sum of all even Fibonacci numbers. See more details in the task description in
+# your assessment paper.
+# """
+#
+def even_fibonacci_sum(limit):
+
+    def fibonacci(n=limit): #note doesn't work but if this worked then rest of function should work
+        fibonacci_list = []
+        # if n <= 2:
+        #     fibonacci(0) == 0
+        #     fibonacci(1) == 1
+        #     fibonacci(2) == 1
+        # else:
+        #     fibonacci(n-2+ n-1)
+            #fibonacci_list.append(fibonacci)
+        return fibonacci_list
+
+    list = fibonacci(limit)
+    sum_total = 0
+
+    for each in list:
+        if each % 2 == 0:
+            sum_total += each
+    return sum_total
+
+# ##### TESTS ####
+#
+# print(even_fibonacci_sum(limit=10))  # should be 44
+# print(even_fibonacci_sum(limit=15))  # should be 188
+# print(even_fibonacci_sum(limit=1))   # should be 0
+#
```

Comment:
Again, you are on the right track here, but you need to look at this in detail again.